### PR TITLE
Update dbt dependencies

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:** https://github.com/fivetran/dbt_zendesk/issues/109

**This PR will result in the following new package version:** v0.12.0

Will require a full refresh to update the logic coming from [dbt-date](https://github.com/calogica/dbt-date/releases)

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

Updates the dbt-date dependency version to the most current 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [ ] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [ ] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

Run with a full refresh

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [ n/a] DECISIONLOG updates have been updated (if applicable)
- [ ] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:2]: Shouldn't this be in it's own `## Updates` section?

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]: Is this intended to be here?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Minor comments in the CHANGELOG. Otherwise, this looks good!

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:2]: true, updated

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:5]: whoops-- removed

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:3]: Can we stick to the format and wording that we use in other packages? 

```suggestion
## Dependency Updates
- The `dbt-date` dependency has been updated to the most current version, [">=0.9.0", "<0.10.0"]. ([PR #113](https://github.com/fivetran/dbt_zendesk/pull/113))
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
One final comment regarding the wording of the CHANGELOG. With that, this is good to go.

Let's plan to fold this change into the other Zendesk changes that are taking place at the moment. As well as holding off on releasing until other dbt-date and expecations updates are ready in other packages.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:192]: This should be <1.0.0, right?

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:4]:
```suggestion
- The `dbt-date` dependency has been updated to the most current version, [">=0.9.0", "<1.0.0"]. ([PR #113](https://github.com/fivetran/dbt_zendesk/pull/113))
```

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Just a few changes to be applied before this will be good to approve. Also, can you resolve the merge conflicts

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:2]: Oops, removed

File: README.md
Comment (fivetran-reneeli) [reply README.md:192]: Yes!

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:4]: updated

**Pull request merged**
